dynosaur/tests/pass/basic-with-self.stdout

```diff
@@ -0,0 +1,23 @@
+#![feature(prelude_import)]
+#[prelude_import]
+use std::prelude::rust_2021::*;
+#[macro_use]
+extern crate std;
+use dynosaur::dynosaur;
+
+trait MyTrait {
+    type Item;
+    async fn foo()
+    -> Self::Item;
+}
+trait ErasedMyTrait {
+    type Item;
+    fn foo<'dynosaur>()
+    ->
+        ::core::pin::Pin<Box<dyn ::core::future::Future<Output = Self::Item> +
+        'dynosaur>>
+    where
+    Self: 'dynosaur;
```

Comment:
It looks like you tried to remove this, but it's still here? Maybe it's seeing `Self::Item` in the signature. But the only thing the where clause would be good for is allowing us to capture arguments in the returned future, and there are no arguments.